### PR TITLE
fix RestController 500 errors

### PR DESCRIPTION
**Pull request by pkit**
This problem exists in both the RestController and the ApiController.

*Controller.load_config was returning too early if the
cluster_config was retrieved from memcached. Later in the
controller call chain, if cluster_config is set, config_path is
expected to be set as well, but the load_config call was returning
prematurely before config_path could be set.

This was resulting in intermittent generic 500 errors when exercising
the /api/ URL REST features. (Basically, a reqest would work the first
time, then throw errors until the cache expires--about 60 seconds
later--and then start working again.)

This patch fixes it by properly initializing config_path at all times
We also verify it by adding a variation of an exsiting test that tests the behavior in memecache presence

### CODE REVIEW

**Comment by pkit** [comment]
addresses #151

File: zerocloud/proxyquery.py [hunk 1]
```diff
@@ -1684,6 +1684,7 @@ def HEAD(self, req):
         return HTTPNotImplemented(request=req)
 
     def load_config(self, req, config_path):
+        self.config_path = config_path
```
Comment (larsbutler) [inline zerocloud/proxyquery.py:1687]: Why are we not caching both `config_path` and `cluster_config` as we are in the `ApiController`?

**Comment by pkit** [comment]
Because in RestController config_path is taken verbatim from request url and not calculated through container metadata (and accessing container metadata could be expensive).

**Comment by larsbutler** [comment]
Alright, cool. Looks good!

**Pull request merged**